src/Kiota.Builder/Writers/Go/CodeProprietableBlockDeclarationWriter.cs

```diff
@@ -17,7 +17,7 @@ public override void WriteCodeElement(T codeElement, LanguageWriter writer)
             writer.WriteLine($"package {ns.Name.GetLastNamespaceSegment().Replace("-", string.Empty)}");
         var importSegments = codeElement
                             .Usings
-                            .Where(x => !x.Declaration.IsExternal && !x.Name.Equals(ns.Name, StringComparison.OrdinalIgnoreCase))
+                            .Where(x => !x.Declaration.IsExternal && !x.Name.Equals(ns?.Name ?? string.Empty, StringComparison.OrdinalIgnoreCase))
```

Comment:
ns being null is a big issue, since we'll be missing the package name in the package directive and go won't compile.
I'd rather investigate why the code namespace is null and address that than fail "safely"

Reply:
that'd make sense. Thanks for looking into that.
What do you think of that for line 15?

```csharp
var ns = codeElement.Parent?.Parent as CodeNamespace ?? codeElement.Parent?.Parent?.Parent as CodeNamespace;
```

Note, the other codepropierableblockdeclaration writers might suffer the same issue. Could you have a look as well please?

Reply:
I'm taking a look at what we are generating now [here](https://github.com/microsoftgraph/msgraph-sdk-go/blob/43e6adf7381e48e873a50cfa0f11da006f76a3ee/chats/count/count_request_builder.go#L18) and making that change would result in us writing a second/double package declaration in for the `CountRequestBuilderGetOptions` declaration as well. 
It might be better to leave as is. Yeah?

Reply:
Agreed.